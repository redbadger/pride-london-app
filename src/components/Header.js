// @flow
import React from "react";
import { StatusBar, StyleSheet, View } from "react-native";
import type { Node } from "react";
import SafeAreaView from "react-native-safe-area-view";
import ContentPadding from "./ContentPadding";
import IconButton from "./IconButton";
import Text from "./Text";
import chevronLeftWhite from "../../assets/images/chevron-left-white.png";
import { lightNavyBlueColor } from "../constants/colors";
import text from "../constants/text";

type Props = {
  onBack?: () => void,
  title?: string,
  rightElement?: Node
};

const renderBackButton = onBack => (
  <IconButton
    accessibilityLabel={text.backButtonAccessibilityLabel}
    onPress={onBack}
    source={chevronLeftWhite}
  />
);

const renderTitle = (title: string) => (
  <Text type="h2" color="whiteColor">
    {title}
  </Text>
);

const Header = ({ onBack, title, rightElement }: Props) => {
  const leftElement = onBack ? (
    renderBackButton(onBack)
  ) : (
    <View style={styles.phantomIcon} />
  );
  const titleElement = title ? renderTitle(title) : null;
  return (
    <SafeAreaView
      accessibilityTraits={["header"]}
      style={{ backgroundColor: lightNavyBlueColor }}
    >
      <StatusBar
        barStyle="light-content"
        animated
        backgroundColor={lightNavyBlueColor}
      />
      <ContentPadding style={styles.headerContent}>
        {leftElement}
        {titleElement}
        {rightElement || <View style={styles.phantomIcon} />}
      </ContentPadding>
    </SafeAreaView>
  );
};

Header.defaultProps = {
  title: undefined,
  onBack: undefined,
  rightElement: undefined
};

const styles = StyleSheet.create({
  headerContent: {
    height: 48,
    flexDirection: "row",
    justifyContent: "space-between",
    alignItems: "center"
  },
  phantomIcon: {
    width: 48,
    height: 48
  }
});

export default Header;
