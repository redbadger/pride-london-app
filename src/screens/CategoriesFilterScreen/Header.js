// @flow
import React from "react";
import { View, StyleSheet } from "react-native";
import Text, { scaleWithFont } from "../../components/Text";
import CommonHeader from "../../components/Header";
import ContentPadding from "../../components/ContentPadding";
import ActionButton from "../../components/ActionButton";
import text from "../../constants/text";
import { whiteColor } from "../../constants/colors";
import CategoriesPills from "../../components/CategoriesPills";
import type { EventCategoryName } from "../../data/event";

type Props = {
  onBack: Function,
  onClearAll?: Function,
  selectedCategories: Set<EventCategoryName>
};

const Header = ({ onBack, onClearAll, selectedCategories }: Props) => (
  <View>
    <CommonHeader
      leftElement={<CommonHeader.BackButton onPress={onBack} />}
      rightElement={
        selectedCategories.size > 0 && (
          <ActionButton label={text.reset} onPress={onClearAll} />
        )
      }
    />
    <ContentPadding>
      <Text type="h1" style={styles.filterTitle}>
        {text.filterTitle}
      </Text>
      <View style={styles.categoriesPills}>
        <CategoriesPills selectedCategories={selectedCategories} />
      </View>
    </ContentPadding>
  </View>
);

Header.defaultProps = {
  onClearAll: () => {}
};

const styles = StyleSheet.create({
  categoriesPills: {
    width: "100%",
    height: scaleWithFont("h3", 44),
    marginBottom: 16
  },
  filterTitle: {
    color: whiteColor,
    marginTop: 4,
    marginBottom: 6
  }
});

export default Header;
