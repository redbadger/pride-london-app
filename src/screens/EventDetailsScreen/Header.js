// @flow
import React from "react";
import type { Node } from "react";
import { StyleSheet, StatusBar, View } from "react-native";
import SafeAreaView from "react-native-safe-area-view";
import { darkBlueGreyTwoColor } from "../../constants/colors";

type Props = {
  children: Node
};

const Header = ({ children }: Props) => (
  <View accessibilityTraits={["header"]} style={styles.header}>
    <SafeAreaView>
      <StatusBar barStyle="light-content" animated />
      {children}
    </SafeAreaView>
  </View>
);

const styles = StyleSheet.create({
  header: {
    backgroundColor: darkBlueGreyTwoColor
  }
});

export default Header;
