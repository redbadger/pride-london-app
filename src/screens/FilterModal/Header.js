// @flow
import React from "react";
import { StyleSheet, View } from "react-native";
import SafeAreaView from "react-native-safe-area-view";
import ActionButton from "../../components/ActionButton";
import Text from "../../components/Text";
import ContentPadding from "../../components/ContentPadding";
import { lightNavyBlueColor, whiteColor } from "../../constants/colors";
import text from "../../constants/text";

type Props = {
  onClearPress: () => void,
  onCancelPress: () => void,
  showClear: boolean
};

const Header = ({ onClearPress, onCancelPress, showClear }: Props) => (
  <SafeAreaView style={styles.container} forceInset={{ top: "always" }}>
    <ContentPadding style={styles.content}>
      <ActionButton label={text.cancel} onPress={onCancelPress} />
      <View style={styles.titleWrapper}>
        <Text
          type="h2"
          style={styles.titleText}
          accessibilityTraits={["header"]}
        >
          {text.filterEvents}
        </Text>
      </View>
      <ActionButton
        label={text.clearAll}
        onPress={onClearPress}
        style={showClear ? styles.show : styles.hide}
        // forces component to re-render after change in style
        key={showClear ? 1 : 0}
      />
    </ContentPadding>
  </SafeAreaView>
);

const styles = StyleSheet.create({
  container: {
    height: 50,
    backgroundColor: lightNavyBlueColor
  },
  content: {
    flex: 1,
    flexDirection: "row",
    justifyContent: "space-between",
    alignItems: "center"
  },
  titleWrapper: { paddingTop: 4 },
  titleText: { color: whiteColor },
  show: { opacity: 1 },
  hide: { opacity: 0 }
});

export default Header;
