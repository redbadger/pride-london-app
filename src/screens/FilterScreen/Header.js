// @flow
import React from "react";
import ActionButton from "../../components/ActionButton";
import GlobalHeader from "../../components/Header";
import text from "../../constants/text";

type Props = {
  onClearPress: () => void,
  onBackPress: () => void,
  showClear: boolean
};

const Header = ({ onClearPress, onBackPress, showClear }: Props) => {
  const clearElement = showClear ? (
    <ActionButton label={text.reset} onPress={onClearPress} />
  ) : null;
  return (
    <GlobalHeader
      leftElement={<GlobalHeader.BackButton onPress={onBackPress} />}
      title={text.filterEvents}
      rightElement={clearElement}
    />
  );
};

export default Header;
