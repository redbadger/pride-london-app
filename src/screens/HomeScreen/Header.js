// @flow
import React from "react";
import { Image, StyleSheet, View } from "react-native";
import type { NavigationScreenProp, NavigationState } from "react-navigation";
import { sortBy } from "ramda";
import homeSupportUsBgBottomLeft from "../../../assets/images/homeSupportUsBgBottomLeft.png";
import homeSupportUsBgTopRight from "../../../assets/images/homeSupportUsBgTopRight.png";
import ButtonWithShapes from "../../components/ButtonWithShapes";
import ContentPadding from "../../components/ContentPadding";
import Text from "../../components/Text";
import { whiteColor, lightNavyBlueColor } from "../../constants/colors";
import { SUPPORT_US } from "../../constants/routes";
import text from "../../constants/text";
import type { FieldRef } from "../../data/field-ref";
import type { ImageSource } from "../../data/get-asset-source";
import type { HeaderBanner } from "../../data/header-banner";

type Props = {
  headerBanners: HeaderBanner[],
  getAssetSource: FieldRef => ImageSource,
  navigation: NavigationScreenProp<NavigationState>
};

const getHeroImageId = (banner: HeaderBanner): string =>
  banner.fields.heroImage.sys.id;

const pickBanner = (banners: HeaderBanner[]): ?HeaderBanner => {
  const sortedBanners = sortBy(getHeroImageId, banners);
  const day = Math.floor(Date.now() / 1000 / 60 / 60 / 24);
  return sortedBanners[day % banners.length];
};

const Header = ({ headerBanners, getAssetSource, navigation }: Props) => {
  const banner = pickBanner(headerBanners);
  if (!banner) return null;

  const heroImage = getAssetSource(banner.fields.heroImage);
  const heroImageHeight = 218; // 225 (header) - 7 (marginTop)
  const heroImageWidth = heroImage.width * (heroImageHeight / heroImage.height);

  return (
    <View>
      <View
        style={[
          styles.container,
          { backgroundColor: banner.fields.backgroundColour }
        ]}
      >
        <Image
          resizeMode="contain"
          source={heroImage}
          style={[
            styles.image,
            { width: heroImageWidth, height: heroImageHeight }
          ]}
        />
        <ContentPadding style={styles.headingContainer}>
          <Text type="uber" color="lightNavyBlueColor" style={styles.heading}>
            {banner.fields.heading}
          </Text>
          {banner.fields.headingLine2 && (
            <Text
              type="uber"
              color="lightNavyBlueColor"
              style={[styles.heading, styles.headingBelow]}
            >
              {banner.fields.headingLine2}
            </Text>
          )}
          {banner.fields.subHeading && (
            <Text type="h2" color="whiteColor" style={styles.subHeading}>
              {banner.fields.subHeading}
            </Text>
          )}
        </ContentPadding>
      </View>
      <ContentPadding style={styles.supportUsButtonContainer}>
        <ButtonWithShapes
          color={lightNavyBlueColor}
          title={text.homeSupportUs}
          description={text.homeSupportUsDescription}
          centerText
          bgBottomLeft={homeSupportUsBgBottomLeft}
          bgTopRight={homeSupportUsBgTopRight}
          navigation={navigation}
          url={SUPPORT_US}
        />
      </ContentPadding>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    height: 225
  },
  image: {
    marginTop: 7,
    marginHorizontal: 10,
    alignSelf: "flex-end"
  },
  headingContainer: {
    position: "absolute",
    height: "100%",
    width: "100%",
    maxWidth: 440,
    alignSelf: "center",
    alignItems: "flex-start",
    justifyContent: "center"
  },
  heading: {
    backgroundColor: whiteColor,
    paddingHorizontal: 8,
    paddingTop: 8,
    zIndex: 2
  },
  headingBelow: {
    marginTop: -6,
    zIndex: 1
  },
  subHeading: {
    backgroundColor: lightNavyBlueColor,
    paddingHorizontal: 8,
    paddingTop: 4
  },
  supportUsButtonContainer: {
    marginTop: -37,
    width: "100%",
    maxWidth: 440,
    alignSelf: "center"
  }
});

export default Header;
