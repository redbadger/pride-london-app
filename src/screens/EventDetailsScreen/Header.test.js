// @flow
import React from "react";
import { shallow } from "enzyme";
import { Text } from "react-native";
import Header from "./Header";

it("renders correctly", () => {
  const output = shallow(
    <Header>
      <Text>Lorem Ipsum</Text>
    </Header>
  );
  expect(output).toMatchSnapshot();
});
