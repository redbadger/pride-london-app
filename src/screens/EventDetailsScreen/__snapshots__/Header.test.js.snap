// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityTraits={
    Array [
      "header",
    ]
  }
  style={
    Object {
      "backgroundColor": "#20215a",
    }
  }
>
  <withOrientation>
    <StatusBar
      animated={true}
      barStyle="light-content"
      showHideTransition="fade"
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Lorem Ipsum
    </Text>
  </withOrientation>
</View>
`;
